apps/crn-server/test/controllers/research-outputs.test.ts

```diff
@@ -596,24 +561,185 @@ describe('ResearchOutputs controller', () => {
       ).rejects.toThrow(NotFoundError);
     });
 
-    test('Should update the research output and return it', async () => {
-      const researchOutputUpdateData = getResearchOutputUpdateData();
+    test('Should throw when trying to unpublish a published research output', async () => {
+      researchOutputDataProviderMock.fetchById.mockResolvedValue({
+        ...getResearchOutputDataObject(),
+        published: true,
+      });
+      researchOutputDataProviderMock.update.mockResolvedValueOnce(
+        researchOutputId,
+      );
+
+      await expect(
+        researchOutputs.update(researchOutputId, {
+          ...getResearchOutputUpdateData(),
+          published: false,
+        }),
+      ).rejects.toThrow(Error);
+    });
+
+    test('Should update a published existing research output and return it', async () => {
```

Comment:
and how to we know that the existing research output is published? Is it that the default fixture has it? well i would expect to be explicit - use the `researchOutputDataProviderMock.fetchById.mockResolvedValue` same way you do above